### ISSUE TITLE
footprints_to_shapes.py and pickpoint failure

### ISSUE THREAD

**Issue by Matioupi** [agisoft-llc/metashape-scripts#10]
Hello,
we are trying to use the footprints_to_shapes.py script, but generated footprints are not correct.

The main case where corners are unproject from model fails for many pictures (we are using a model generated from a very sparse point cloud ~ 75 m step with a bounding box that is much wider than the project bouding box as we just need rough footprints, so we would expect that pickpoint is always able to return a ray intersection.)

The fallover case were corners are unprojected from point_cloud do not fai (always return a value)l, but results seems wrong and the resulting footprints do not macth reality at all.

Do you have more details about the pickPont API for both model and dense_cloud ?

Regards

**Comment by PolarNick239** [comment]
Hello,

pickPoint from model returns the first intersection from all ray intersections with triangles of model. So if ray (corresponding to corner of photo) doesn't intersect model - pickPoint resturns None.

pickPoint from points cloud projects all tie points on ray (corresponding to corner of photo) and returns from them the first one (i.e. the projection closest to ray origin, i.e. closest to camera position).

So on outskirts of the dataset where corners of photo doesn't have tie points nearly - far tie point will be used - it will be projected on ray (corresponding to the corner) and the projection can be far from real surface.

**Comment by Matioupi** [comment]
Thank you for additionnal details.
The background of our request is that we try to generate footprints from only the a priori orientations, and a model generated from a sparse (county scale) point cloud.
So we have a first script that "force" the orientation of the pictures without actually aligning them (no tie points generated) and the second script to generate the footprints from this repo.
If i understand correctly, as we have no tie points, the fall case to pointcloud pickPoint basically returns "non sense" values.
Still when generating a model from this sparse point cloud (actually a DSM), I don't understand why the model pickpoint would fail as the model is much wider than what is actually needed.

**Comment by PolarNick239** [comment]
Yes, if the model is big enough and all rays intersect it - tie points cloud shouldn't be used.

Can you please provide the dataset (not necessary full - you can try to minimize the project such that problem still will be easy to reproduce, and send it as .psz project) to support@agisoft.com for internal testing?

**Comment by Matioupi** [comment]
![image](https://user-images.githubusercontent.com/5566194/63595653-c4a0b980-c5b9-11e9-9319-9f2dcd64eb87.png)
We sent you some data, but this picture is to further illustrate a very light fake project.
I just took 4 images (actually 4 time the same image) with 0 0 0 orientations and position 20m one from each other, above a flat ground ~200m below.
The obtained footprint obvioulsy have an issue and I have no clue where it comes from

**Comment by PolarNick239** [comment]
Can you please send this light fake project and tell how you created these four cameras (if with python script - please provide it)?

In your big project footprints seems to work (model created from tie points cloud):

![test](https://user-images.githubusercontent.com/1218605/63606795-8d420500-c5d9-11e9-84c4-e12c5c5fa7ff.jpg)

**Comment by germanwilches** [comment]
Hola yo he intentado usar el script footprints_to_shapes.py con buenos resultados con pocas fotos, pero cuando intento hacerlo en un proyecto con 15000 imágenes el programa no responde y se cuelga.
